app/kuma-dp/pkg/dataplane/dnsserver/dnsserver.go

```diff
@@ -87,6 +92,7 @@ func (s *DNSServer) NeedLeaderElection() bool {
 }
 
 func (s *DNSServer) Start(stop <-chan struct{}) error {
+	s.finalizer.Running()
```

Comment:
Kind of a nit but maybe `Running()` -> `Start()`, since it's called in `Start` and `Running()` could easily also be the name of `IsRunning() bool` kind of method

Reply:
finalizer is gone